io/src/real_sense/librealsense/real_sense_grabber.cpp

```diff
@@ -0,0 +1,310 @@
+        if ((color_width == color_width_old && color_fps == color_fps_old) || (depth_fps != color_fps)) continue;
+        color_width_old = color_width;
+        color_fps_old = color_fps;
+        Mode mode;
+        mode.fps = depth_fps;
+        mode.depth_width = depth_width;
+        mode.depth_height = depth_height;
+        mode.color_width = color_width;
+        mode.color_height = color_height;
+        modes.push_back (mode);
+      }
+    }
+    else
+    {
+      Mode mode;
+      mode.fps = depth_fps;
+      mode.depth_width = depth_width;
+      mode.depth_height = depth_height;
+      modes.push_back (mode);
+    }
+  }
+  return modes;
+}
+
+void
+pcl::RealSenseGrabber::run ()
+{
+  rs::device* device = device_->getDevice ();
+  device->start ();
+  int depth_width = mode_selected_.depth_width, depth_height = mode_selected_.depth_height, 
+      depth_size = depth_width * depth_height, color_width = mode_selected_.color_width, 
+      color_height = mode_selected_.color_height, color_size = color_width * color_height * 3;
+  ///Buffer to store depth data
+  std::vector<uint16_t> depth_data;
+  depth_data.resize (depth_size);
+  ///Buffer to store color data
+  std::vector<uint8_t> color_data;
+  color_data.resize (color_size);
+  while (is_running_)
+  {
+    if (device->is_streaming ()) device->wait_for_frames ();
+    fps_mutex_.lock ();
+    frequency_.event ();
+    fps_mutex_.unlock ();
+    // Retrieve our images
+    const uint16_t * depth_image = (const uint16_t *)device->get_frame_data (rs::stream::depth);
+    const uint8_t * color_image = (const uint8_t *)device->get_frame_data (rs::stream::color);
+    // Retrieve camera parameters for mapping between depth and color
+    rs::intrinsics depth_intrin = device->get_stream_intrinsics (rs::stream::depth);
+    rs::intrinsics color_intrin = device->get_stream_intrinsics (rs::stream::color);
+    rs::extrinsics depth_to_color = device->get_extrinsics (rs::stream::depth, rs::stream::color);
+    float scale = device->get_depth_scale ();
+    depth_data.clear ();
+    pcl::PointCloud<pcl::PointXYZ>::Ptr xyz_cloud;
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr xyzrgba_cloud;
+    float max_distance = 6; // get rid of noisy data that is past 6 meters
+    static const float nan = std::numeric_limits<float>::quiet_NaN ();
+
+    memcpy (depth_data.data (), &depth_image[0], depth_size * sizeof (uint16_t));
+
+    if (need_xyzrgba_)
+    {
+      color_data.clear ();
+      memcpy (color_data.data (), &color_image[0], color_size * sizeof (uint8_t));
+      xyzrgba_cloud.reset (new pcl::PointCloud<pcl::PointXYZRGBA> (depth_width, depth_height));
+      xyzrgba_cloud->is_dense = false;
+      if (need_xyz_)
+      {
+        xyz_cloud.reset (new pcl::PointCloud<pcl::PointXYZ> (depth_width, depth_height));
+        xyz_cloud->is_dense = false;
+      }
+      for (int dy = 0; dy < depth_height; ++dy)
+      {
+        uint i = dy * depth_width - 1;
```

Comment:
Why do you use <code>uint</code> instead of <code>uint32_t</code>? (#L215, #L280)